### PR TITLE
feat: implement Validate()

### PR DESCRIPTION
**Pull request by sailorlqh**

### CODE REVIEW

File: pkg/plugin/plugin.go [hunk 1]
```diff
@@ -16,17 +16,75 @@ package plugin
...
+	githubCategory               = "github"
+	respAnnotationKeyIssueOwner  = "github_issue_owner"
+	respAnnotationKeyIssueRepo   = "github_issue_repo"
+	respAnnotationKeyIssueNumber = "github_issue_number"
+)
 
-func (g *GithubPlugin) Validate(ctx context.Context, req *jvspb.ValidateJustificationRequest) (*jvspb.ValidateJustificationResponse, error) {
-	return nil, fmt.Errorf("Unimplemented")
+// issueMatcher is the mockable interface for the convenience of testing.
+type issueMatcher interface {
```
Comment (sqin2019) [inline pkg/plugin/plugin.go:40]: This should live in validator class.

File: pkg/plugin/plugin.go [hunk 2]
```diff
@@ -16,17 +16,75 @@ package plugin
...
+	MatchIssue(ctx context.Context, issueURL string) (*pluginGitHubIssue, error)
+}
+
+// GithubPlugin is the implementation of jvspb.Validator interface.
```
Comment (sqin2019) [inline pkg/plugin/plugin.go]: nit: jvspb.Validator link reference

File: pkg/plugin/plugin.go [hunk 3]
```diff
@@ -16,17 +16,75 @@ package plugin
...
+type GitHubPlugin struct {
+	validator issueMatcher
```
Comment (sqin2019) [inline pkg/plugin/plugin.go:49]: nit: comment

File: pkg/plugin/plugin_test.go [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+	"testing"
+
+	jvspb "github.com/abcxyz/jvs/apis/v0"
+	"github.com/abcxyz/pkg/testutil"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+)
+
+type testIssueMatcher struct {
+	rPluginGithubIssue *pluginGitHubIssue
```
Comment (sqin2019) [inline pkg/plugin/plugin_test.go]: nit: GitHub through out

File: pkg/plugin/plugin.go [hunk 5]
```diff
@@ -16,17 +16,75 @@ package plugin
...
+		} else {
+			return nil, status.Errorf(codes.Internal, err.Error())
+		}
+	}
+	return &jvspb.ValidateJustificationResponse{
+		Valid: true,
+		Annotation: map[string]string{
+			respAnnotationKeyIssueOwner:  info.Owner,
+			respAnnotationKeyIssueRepo:   info.RepoName,
+			respAnnotationKeyIssueNumber: strconv.Itoa(info.IssueNumber),
```
Comment (sqin2019) [inline pkg/plugin/plugin.go:79]: Maybe put the original link too.

File: pkg/plugin/plugin_test.go [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+			t.Parallel()
+
+			p := &GitHubPlugin{
+				validator: tc.validator,
+			}
+			gotResq, gotErr := p.Validate(ctx, tc.req)
+			if diff := testutil.DiffErrString(gotErr, tc.wantErr); diff != "" {
+				t.Errorf(diff)
+			}
+			if diff := cmp.Diff(tc.wantResq, gotResq, cmpopts.IgnoreUnexported(jvspb.ValidateJustificationResponse{})); diff != "" {
```
Comment (sqin2019) [inline pkg/plugin/plugin_test.go:142]: with cmpopts.IgnoreUnexported(jvspb.ValidateJustificationResponse{}), does it compare ValidateJustificationResponse?

File: pkg/plugin/plugin.go [hunk 7]
```diff
@@ -16,17 +16,75 @@ package plugin
...
 }
 
-func (g *GithubPlugin) GetUIData(ctx context.Context, req *jvspb.GetUIDataRequest) (*jvspb.UIData, error) {
+// Validate returns the validation result.
+func (g *GitHubPlugin) Validate(ctx context.Context, req *jvspb.ValidateJustificationRequest) (*jvspb.ValidateJustificationResponse, error) {
+	if got, want := req.Justification.Category, githubCategory; got != want {
+		return generateInvalidErrResq(fmt.Sprintf("failed to perform validation, expected category %q to be %q", got, want)), nil
+	}
+
+	info, err := g.validator.MatchIssue(ctx, req.Justification.Value)
```
Comment (sqin2019) [inline pkg/plugin/plugin.go:65]: What if req.Justification.Value is empty, does it return invalidJustification error as we expected? maybe we should add a check on this specifically.

Comment (sailorlqh) [reply pkg/plugin/plugin.go:65]: Yes, if it's empty, it won't pass the regexp check, so I will return invalidJustification error.

File: pkg/plugin/plugin_test.go (hunk 6 above)
Comment (sailorlqh) [reply pkg/plugin/plugin_test.go:142]: Yes, it only ignore the unexported field in `jvspb.ValidateJustificationResponse`, the rest of the fields will still be compared.

File: pkg/plugin/plugin.go (hunk 7 above)
Comment (sqin2019) [reply pkg/plugin/plugin.go:65]: Ah, forgot about that, we are good here than, thanks!

File: pkg/plugin/plugin_test.go (hunk 6 above)
Comment (sqin2019) [reply pkg/plugin/plugin_test.go:142]: I see, thanks!

File: pkg/plugin/plugin.go (hunk 1 above)
Comment (sqin2019) [reply pkg/plugin/plugin.go:40]: Sorry for the churn, after a second look, since issueMatcher only used in plugin.go, please ignore this comment.

**Review by sqin2019 — DISMISSED** [review]

**Review by sqin2019 — APPROVED** [review]

**Review by sethvargo — APPROVED** [review]

**Pull request merged**